pynitrokey/cli/storage.py

```diff
@@ -6,27 +6,85 @@
 # http://apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
 # http://opensource.org/licenses/MIT>, at your option. This file may not be
 # copied, modified, or distributed except according to those terms.
-
+import logging
+import platform
+import subprocess
 
 import click
 
 from pynitrokey.helpers import local_critical, local_print
 from pynitrokey.libnk import BaseLibNitrokey, DeviceNotFound, NitrokeyStorage
 
+logger = logging.getLogger(__name__)
 
 @click.group()
 def storage():
     """(experimental) 'Nitrokey Storage' keys, see subcommands."""
     pass
 
 
+@click.command()
+@click.argument('firmware', type=click.Path(exists=True, readable=True))
+@click.option(
+    "--experimental",
+    default=False,
+    is_flag=True,
+    help="Allow to execute experimental features",
+)
+def update(firmware: str, experimental):
+    """experimental: run assisted update through dfu-programmer tool"""
+    if platform.system() != "Linux" or not experimental:
+        local_print("This feature is Linux only and experimental, which means it was not tested thoroughly.\n"
+                    "Please pass --experimental switch to force running it anyway.")
+        raise click.Abort()
+    assert firmware.endswith('.hex')
+
+    commands = f"""
+        sudo dfu-programmer at32uc3a3256s erase
+        sudo dfu-programmer at32uc3a3256s flash --suppress-bootloader-mem {firmware}
```

Comment:
Couldn’t this cause issues if `firmware` contains a space?

Reply:
Indeed, good catch - to be corrected

Reply:
Corrected in latest update